### PR TITLE
fix 3685 and 4093  - business create

### PR DESCRIPTION
**Pull request by peter-freshworks**
*Issue #:* /bcgov/entity#3685
*Issue #:* /bcgov/entity#4093
*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

File: src/mixins/bcol-mixin.ts [hunk 1]
```diff
@@ -4,24 +4,28 @@ import axios from '@/utils/axios-auth'
...
+      const fetchUrl: string = this.payApi + 'codes/errors/' + errCode
+      // Currently no desirable way to handle errors during this request,
+      // null is returned in any error situation regardless.
+      const errObj: any = await axios.get(fetchUrl).catch()
+      if (errObj?.data) {
+        return errObj.data
+      }
+      return null
+    } catch (error) {
+      return null
```
Comment (cameron-freshworks) [inline src/mixins/bcol-mixin.ts:17]: Just curious if you need the try catch if `null` is return regardless. Are you able to `return errObj?.data || null` or are you expecting a potential error on the `fetchUrl`?

File: src/mixins/bcol-mixin.ts [hunk 2]
```diff
@@ -4,24 +4,28 @@ import axios from '@/utils/axios-auth'
...
     }
-    return null
   }
 
   getErrorCode (error: any): string {
     if (error?.response?.data?.errors) {
-      const msgCode = error.response.data.errors.find(x => x.payment_error_type.startsWith('BCOL'))
+      const msgCode = error.response.data.errors.find(x => x.payment_error_type && x.payment_error_type.startsWith('BCOL'))
```
Comment (cameron-freshworks) [inline src/mixins/bcol-mixin.ts:23]: Does optional chaining work here?

File: src/App.vue [hunk 3]
```diff
@@ -262,6 +262,7 @@ export default class App extends Mixins(BcolMixin, DateMixin, FilingTemplateMixi
       // process errors/warnings
       switch (error?.response?.status) {
         case PAYMENT_REQUIRED:
+          this.haveChanges = false
```
Comment (severinbeauvais) [inline src/App.vue:265]: Please add a comment why it's OK to clear this flag in this situation.

Comment (severinbeauvais) [reply src/App.vue:265]: (The principle I'm following is that code paths are often symmetric/similar, and in this case they're not. As a reviewer, I wonder why they're not. Totally fair scenario, just document it.)

File: src/mixins/bcol-mixin.ts (hunk 1 above)
Comment (severinbeauvais) [reply src/mixins/bcol-mixin.ts:17]: I'm wondering the same thing.
Also, if you get a lint error about "error not used" then you can change this to the special form `catch (e)` (where it ignores the fact that you're not using e).

File: src/mixins/bcol-mixin.ts (hunk 2 above)
Comment (peter-freshworks) [reply src/mixins/bcol-mixin.ts:23]: it does

File: src/App.vue (hunk 3 above)
Comment (severinbeauvais) [reply src/App.vue:266]: Perfect, thanks!

File: src/mixins/bcol-mixin.ts [hunk 4]
```diff
@@ -8,7 +8,7 @@ export default class BcolMixin extends Vue {
       const fetchUrl: string = this.payApi + 'codes/errors/' + errCode
       // Currently no desirable way to handle errors during this request,
       // null is returned in any error situation regardless.
-      const errObj: any = await axios.get(fetchUrl).catch()
+      const errObj: any = await axios.get(fetchUrl)
```
Comment (severinbeauvais) [inline src/mixins/bcol-mixin.ts:11]: Have you tested the scenario where this get fails (exception)? Easy to do by corrupting the fetchUrl string above. This just to make sure the error is handled correctly.

File: src/mixins/bcol-mixin.ts [hunk 5]
```diff
@@ -20,7 +20,9 @@ export default class BcolMixin extends Vue {
 
   getErrorCode (error: any): string {
     if (error?.response?.data?.errors) {
-      const msgCode = error.response.data.errors.find(x => x.payment_error_type && x.payment_error_type.startsWith('BCOL'))
+      const msgCode = error.response.data.errors.find(x =>
+        x?.payment_error_type?.startsWith('BCOL')) // eslint-disable-line camelcase
```
Comment (severinbeauvais) [inline src/mixins/bcol-mixin.ts:24]: FYI, another way to add the lint exception for a long line is to specify is before the line with the command "eslint-disable-next-line ...".

**Comment by severinbeauvais** [comment]
Although the code could be improved, I don't see any issue with merging this anyway.

**Pull request merged**